sdk/spring/azure-spring-boot-test-aad/HOW_TO_RUN_AAD_INTEGRATION_TEST_IN_LOCALHOST.md

```diff
@@ -0,0 +1,116 @@
+# How to run AAD integration test in localhost
+
+## Create environment variables
+
+Here is the list of all required environment variables:
+```
+AAD_TENANT_ID_1
+AAD_USER_NAME_1
+AAD_USER_PASSWORD_1
+AAD_TENANT_ID_2
+AAD_USER_NAME_2
+AAD_USER_PASSWORD_2
+AAD_MULTI_TENANT_CLIENT_ID
+AAD_MULTI_TENANT_CLIENT_SECRET
+AAD_SINGLE_TENANT_CLIENT_ID
+AAD_SINGLE_TENANT_CLIENT_SECRET
+AAD_SINGLE_TENANT_CLIENT_ID_WITH_ROLE
+AAD_SINGLE_TENANT_CLIENT_SECRET_WITH_ROLE
+```
+
+Let's go through this document to get all these variables.
+
+### `AAD_TENANT_ID_1`
+
+The first tenant-id we need for AAD test.
+
+1. Create a new tenant named `aadIntegrationTest1`. Refs: [Create a new tenant in Azure Active Directory][Create a new tenant in Azure Active Directory].
+
+2. After the tenant created successfully. Click `Tenant creation was successful. Click here to navigate to your new tanent: aadIntegrationTest1`.
+![created-a-new-tenant](images/created-a-new-tenant.png)
+
+3. Copy `Tenant ID`. Then we get the value of `AAD_TENANT_ID_1`.
+![copy-tenant-id](images/copy-tenant-id.png)
+
+### `AAD_USER_NAME_1`, `AAD_USER_PASSWORD_1`
+
+1. Create a new user named `user1`. Refs: [Add or delete users using Azure Active Directory][Add or delete users using Azure Active Directory].
+
+2. Copy the user name and initial password. In the following image, user name = `user1@aadIntegrationTest1.onmicrosoft.com`, initial password = `Xabo2448`.
+![create-user1](images/create-user1.png)
+
+3. In Edge, open an InPrivate window to change password for new created user.
+![open-inPrivate-window](images/open-inPrivate-window.png)
+![change-password](images/change-password.png)
+
+4. Now we get the value of `AAD_USER_NAME_1`, `AAD_USER_PASSWORD_1`.
+
+### `AAD_TENANT_ID_2`, `AAD_USER_NAME_2`, `AAD_USER_PASSWORD_2` 
+Get `AAD_TENANT_ID_2`, `AAD_USER_NAME_2`, `AAD_USER_PASSWORD_2` with the same method.
+
+### `AAD_MULTI_TENANT_CLIENT_ID`, `AAD_MULTI_TENANT_CLIENT_SECRET`.
```

Comment:
And this multi-tenant application will be created in which tenant?

Reply:
Updated to
```
Register an application in the new created tenant (`aadIntegrationTest1`).
```